Client/Frontend/Browser/RecentlyClosedTabs.swift

```diff
@@ -0,0 +1,97 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+public protocol RecentlyClosedTabs {
+    var tabs: [ClosedTab] { get }
+    var count: Int { get }
+    func addTab(url: NSURL, title: String?, faviconURL: String?)
+    func clearTabs()
+}
+
+public class RecentlyClosedTabsStore: RecentlyClosedTabs {
+    lazy public var tabs: [ClosedTab] = {
+        guard let recentlyClosedTabs = NSUserDefaults.standardUserDefaults().objectForKey("recentlyClosedTabs") else {
```

Comment:
Don't use standard user defaults, use the profile prefs. The reason I suggested moving it into the profile in the first place was so that the prefs would be profile-specific.